core/includes/menu.inc

```diff
@@ -489,6 +491,43 @@ function menu_get_item($path = NULL, $router_item = NULL) {
   return $router_items[$path];
 }
 
+function menu_get_path_wildcards($string) {
+  $results = array();
+  $array = explode('/', $string);
+  $first_char = array_shift($array);
+  $count = count($array);
+  $max = bindec(str_pad('', $count, '1'));
```

Comment:
Fancy! 😅

Reply:
Not mine. https://backdrop.zulipchat.com/#narrow/stream/228714-Community-Chat.20.28other.20topics.29/topic/Array.20permutation.20algorithm/near/196272997